overlord/snapstate/snapstate.go

```diff
@@ -457,6 +457,8 @@ var CheckHealthHook = func(st *state.State, snapName string, rev snap.Revision)
 	panic("internal error: snapstate.CheckHealthHook is unset")
 }
 
+var generateSnapdWrappers = backend.GenerateSnapdWrappers
+
 // WaitRestart will return a Retry error if there is a pending restart
 // and a real error if anything went wrong (like a rollback across
 // restarts)
```

Comment:
maybe call this now FinishRestart and then add to the comment:

/ ... 
// For snapd snap updates this will also rerun wrappers generation to fully catch up with any change.